enterprise/core-edge/src/test/java/org/neo4j/coreedge/raft/log/segmented/TermsTest.java

```diff
@@ -276,16 +276,35 @@ public void shouldAppendAfterSkip() throws Exception
         assertTermInRange( skipIndex + 1, skipIndex + 20, skipTerm );
     }
 
+    @Test
+    public void shouldMaintainSaneMemoryUseForTerms() throws Exception
```

Comment:
You could expose the size for the benefit of the test, and assert on it shrinking. That should lead to tests that are more on-point.